Pokecon/Program.cs

```diff
@@ -85,22 +85,122 @@ static ResponseEnvelop api_req(string api_endpoint, string access_token, Request
             }
         }
 
-        static string get_api_endpoint(string access_token)
+        static string get_api_endpoint(string access_token, string provider)
         {
             var ret = api_req(API_URL, access_token, new[] {
                 new RequestEnvelop.Types.Requests { Type = 2 },
                 new RequestEnvelop.Types.Requests { Type = 126 },
                 new RequestEnvelop.Types.Requests { Type = 4 },
                 new RequestEnvelop.Types.Requests { Type = 129 },
                 new RequestEnvelop.Types.Requests { Type = 5, Message = new RequestEnvelop.Types.Unknown3 { Unknown4 = "4a2e9bc330dae60e7b74fc85b98868ab4700802e" } },
-            });
+            }, provider);
             try { return "https://" + ret.ApiUrl + "/rpc"; }
-            catch { return null; }
+            catch (Exception e){
+                return null;
+            }
+        }
+
+        static ResponseEnvelop get_profile(string api_endpoint, string access_token, string provider)
+        {
+            return api_req(api_endpoint, access_token, new[] { new RequestEnvelop.Types.Requests { Type = 2 } }, provider);
         }
 
-        static ResponseEnvelop get_profile(string api_endpoint, string access_token)
+        static string login_google(string username, string password)
         {
-            return api_req(api_endpoint, access_token, new[] { new RequestEnvelop.Types.Requests { Type = 2 } });
+            Console.WriteLine("[!] Google login for: {0}", username);
+            var first = "https://accounts.google.com/o/oauth2/auth?client_id=848232511240-73ri3t7plvk96pj4f85uj8otdat2alem.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&response_type=code&scope=openid%20email%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email";
+            var second = "https://accounts.google.com/AccountLoginInfo";
+            var third = "https://accounts.google.com/signin/challenge/sl/password";
+            var last = "https://accounts.google.com/o/oauth2/token";
+            using (var clientHandler = new HttpClientHandler())
+            {
+                clientHandler.AllowAutoRedirect = true;
+                using (var client = new HttpClient(clientHandler))
+                {
+                    client.DefaultRequestHeaders.Add("User-Agent", "Mozilla/5.0 (iPad; CPU OS 8_4 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) Mobile/12H143");
+                    var response = client.GetAsync(first).Result;
+                    var r = response.Content.ReadAsStringAsync().Result;
+
+                    var galx_regex = "name=\"GALX\" value=\"(.*?)\"";

```

Comment:
Why have 3 different expressions which are all basically the same except the name?

You could use `name="(\w+?)" value="(.*?)"`